docs/modules/data-structures/pages/listening-for-map-entries.adoc

```diff
@@ -36,7 +36,18 @@ C++::
 --
 [source,cpp]
 ----
-
+#include <hazelcast/client/hazelcast_client.h>
+
+class Employee {
+public:
+    std::string surname;
```

Comment:
make it private or make this class a struct and you do not even need the constructor.

Reply:
this is how they implemented in java. I guess it is done that way to imitate a real world application